Src/Tests/Temporal.Sdk.Common.Tests/Serialization/TestUnnamedContainerPayloadConverter.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using Temporal.Api.Common.V1;
+using Temporal.Common.Payloads;
+using Temporal.Serialization;
+using Xunit;
+
+namespace Temporal.Sdk.Common.Tests.Serialization
+{
+    public class TestUnnamedContainerPayloadConverter
+    {
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_String()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            Payloads p = new Payloads();
+            Assert.False(instance.TrySerialize(String.Empty, p));
+            Assert.Empty(p.Payloads_);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Null()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            Payloads p = new Payloads();
+            Assert.False(instance.TrySerialize<string>(null, p));
+            Assert.Empty(p.Payloads_);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Unnamed_SerializedDataBacked()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            instance.InitDelegates(new[] { new JsonPayloadConverter() });
+            Payloads p = new Payloads();
+            NewtonsoftJsonPayloadConverter converter = new NewtonsoftJsonPayloadConverter();
+            converter.Serialize(new SerializableClass { Name = "test", Value = 2 }, p);
+            PayloadContainers.Unnamed.SerializedDataBacked data = new PayloadContainers.Unnamed.SerializedDataBacked(p, converter);
+            Assert.True(instance.TrySerialize(data, p));
+            Assert.NotEmpty(p.Payloads_);
+            Assert.True(instance.TryDeserialize(p, out PayloadContainers.Unnamed.SerializedDataBacked cl));
+            Assert.NotNull(cl);
+            SerializableClass deserializedData = cl.GetValue<SerializableClass>(0);
```

Comment:
Let's also check that `cl.PayloadConverter` is set to `instance`.

Reply:
I added the check. `cl.PayloadConverter` is actually equal to `CompositePayloadConverter`. This is correct behavior, but it seems weird until you realize the `UnnamedContainerPayloadConverter` uses the `CompositePayloadConverter` behind the scenes.

Reply:
I modified that test case to account for this in a later PR.